src/qforte/circuit.cc

```diff
@@ -226,6 +226,32 @@ bool Circuit::is_pauli() const {
     return true;
 }
 
+double Circuit::get_phase_gate_parameter(const Gate& gate) {
+    std::unordered_map<GateType, double> gate_parameters = {
+        {GateType::T, M_PI / 4},
+        {GateType::S, M_PI / 2},
+        {GateType::Z, M_PI}
+    };
+
+    if (gate.has_parameter()) {
+        return gate.parameter().value();
+    } else {
+        auto it = gate_parameters.find(gate.gate_type());
+        if (it != gate_parameters.end()) {
+            return it->second;
+        } else {
+            throw std::invalid_argument("Unknown single-qubit phase gate encountered: " + gate.gate_id());
+        }
+    }
+}
+
+double Circuit::simplify_phase_1qubit_gates(const Gate& gate1, const Gate& gate2) {
```

Comment:
Not wrong. However, the idea to create a method should also be general, meaning that this method should be called for various purposes. I don't see the necessity to have a function here which is just a double addition. You can add the params directly and leave a comment there for clarification.